libs/common/CMakeLists.txt

```diff
@@ -33,3 +33,36 @@ target_link_libraries(libs_timeout INTERFACE
   common
   rxcpp
   )
+
+add_library(irohad_version irohad_version.cpp)
+
+# Get the git repo data
+set(GIT_REPO_PRETTY_VER "version info unavailable")
+if (EXISTS "${PROJECT_SOURCE_DIR}/.git")
+  if(NOT GIT_FOUND)
+    find_package(Git QUIET)
+  endif()
+
+  if(GIT_EXECUTABLE)
+    # Get pretty version
+    execute_process(COMMAND
+        "${GIT_EXECUTABLE}"
+        describe --always
+        WORKING_DIRECTORY
+        "${PROJECT_SOURCE_DIR}"
+        RESULT_VARIABLE
+        res
+        OUTPUT_VARIABLE
+        GIT_REPO_PRETTY_VER
+        ERROR_QUIET
+        OUTPUT_STRIP_TRAILING_WHITESPACE)
+    if(NOT res EQUAL 0)
+      message(STATUS "Error running `git describe': ${res}")
```

Comment:
I suggest in error cases initialize `irohad_version` with "unknown version" or something similar.

Reply:
it is actually done here:
https://github.com/hyperledger/iroha/blob/404c95243536034642b8ac9a36146be8ce7426af/libs/common/CMakeLists.txt#L40